var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -37,61 +42,67 @@ class ModelHippocampus(SimModel):
 
     phases = ('build', 'install')
 
-    mech_name = ''
+    mech_name = None
     """The name of the mechanism, defined in subclasses"""
 
     def build(self, spec, prefix):
-        """Build phase
-        """
-        link_flag = '-Wl,-rpath,' + prefix.lib
-        self._build_mods('mod', link_flag)
+        """Build phase"""
+        self._build_mods('mod')
+
+    @property
+    def lib_suffix(self):
+        return ('_' + self.mech_name) if self.mech_name else ''
 
     def _build_mods(self, mods_location, link_flag='', include_flag='', corenrn_mods=None):
         """Build shared lib & special from mods in a given path
         """
-        profile_flag = ' -DENABLE_TAU_PROFILER' if '+profile' in self.spec else ''
-        include_flag += profile_flag
+        # pass include and link flags for all dependency libraries
+        # Compiler wrappers are not used to have a more reproducible building
+        dep_names = set(self.spec.dependencies_dict('link').keys())
+        for dep in dep_names:
+            if self.spec[dep].prefix in ('/usr', '/usr/local'):
+                raise Exception('Dependency lib "' + dep + '" coming from system prefix')
+        link_flag += ' ' + ' '.join(self.spec[dep].libs.ld_flags for dep in dep_names)
+        link_flag += ' ' + ' '.join(self.spec[dep].libs.rpath_flags for dep in dep_names)
+        include_flag += ''.join(' -I' + str(self.spec[dep].prefix.include) for dep in dep_names)
+        include_flag += ' -DENABLE_TAU_PROFILER' if '+profile' in self.spec else ''
+        output_dir = os.path.basename(self.neuron_archdir)
 
         if self.spec.satisfies('+coreneuron'):
-            mechlib = self.__build_mods_coreneuron(
-                corenrn_mods or mods_location, link_flag, include_flag)
-            # Link neuron special with this mechs lib
-            link_flag += ' %s %s' % (mechlib.ld_flags, self._get_link_flags('coreneuron'))
+            libnrncoremech = self.__build_mods_coreneuron(
+                corenrn_mods or mods_location, link_flag, include_flag
+            )
+            # Relevant flags to build neuron's nrnmech lib
             include_flag += ' -DENABLE_CORENEURON'  # only now, otherwise mods assume neuron
+            include_flag += ' -I%s' % self.spec['coreneuron'].prefix.include
```

Comment:
Could this not just be string concatenation?

Reply:
I'm taking line length seriously now :) and splitting concatenation in some lines is always ugly... it's always ugly anyway, so hope you dont mind this choice